src/main/java/org/apache/commons/collections4/bloomfilter/CountingBloomFilter.java

```diff
@@ -229,47 +229,47 @@ default boolean merge(final Hasher hasher) {
     }
 
     /**
-     * Merges the specified index producer into this Bloom filter.
+     * Merges the specified index extractor into this Bloom filter.
      *
-     * <p>Specifically: all unique cells for the indices identified by the {@code indexProducer} will be incremented by 1.</p>
+     * <p>Specifically: all unique cells for the indices identified by the {@code indexExtractor} will be incremented by 1.</p>
      *
      * <p>This method will return {@code true} if the filter is valid after the operation.</p>
      *
-     * <p>Note: If indices that are returned multiple times should be incremented multiple times convert the IndexProducer
-     * to a CellProducer and add that.</p>
+     * <p>Note: If indices that are returned multiple times should be incremented multiple times convert the IndexExtractor
+     * to a CellExtractor and add that.</p>
      *
-     * @param indexProducer the IndexProducer
+     * @param indexExtractor the IndexExtractor
      * @return {@code true} if the removal was successful and the state is valid
      * @see #isValid()
-     * @see #add(CellProducer)
+     * @see #add(CellExtractor)
      */
     @Override
-    default boolean merge(final IndexProducer indexProducer) {
-        Objects.requireNonNull(indexProducer, "indexProducer");
+    default boolean merge(final IndexExtractor indexExtractor) {
+        Objects.requireNonNull(indexExtractor, "indexExtractor");
         try {
-            return add(CellProducer.from(indexProducer.uniqueIndices()));
+            return add(CellExtractor.from(indexExtractor.uniqueIndices()));
         } catch (final IndexOutOfBoundsException e) {
             throw new IllegalArgumentException(
```

Comment:
This assumes a lot of the implementations. I think it is OK to just Javadoc the `IndexOutOfBoundsException` in the implementation, or, have the implementations convert whatever error condition into an `IllegalArgumentException `. There Javadoc _here_ can state that it is expected that an implementation throws `IllegalArgumentException` and not other exceptions on bad input.